### PR TITLE
replaced design response R code with python implementation

### PR DESCRIPTION
**Pull request by tymor22**
Replacing design response R code with a python implementation. Add test cases specific to python, remove R code and remove Rdriver file used to execute R code.

### CODE REVIEW

File: inferelator_ng/tests/test_design_response.py [hunk 1]
```diff
@@ -20,21 +22,52 @@ def calculate_design_and_response(self):
...
+        expression_3 = np.array(list(self.exp['ts3' + spchrs]))
+        expected_response_2 = expression_2 + self.tau * (expression_3 - expression_2) /  (float(self.meta['del.t'][2]))
+        np.testing.assert_almost_equal(np.array(resp['ts1' + spchrs]), expected_response_1)
+        np.testing.assert_almost_equal(np.array(resp['ts2' + spchrs]), expected_response_2)
+
+
 class TestDRModelOrganisms(TestDR):
 
     def test_on_bsubtilis(self):
-        self.exp = utils.df_from_tsv('data/bsubtilis/expression.tsv')
```
Comment (NicoDeVeaux) [inline inferelator_ng/tests/test_design_response.py]: @tymor22 This is why there's an error in the Travis CI tests : you should change these 4 lines back

**Review by NicoDeVeaux — COMMENTED** [review]
This is also the reason the Travis tests are still failing (IOError: File /Users/tymorhamamsy/inferelator_ng/data/bsubtilis/expression.tsv does not exist
)

**Comment by NicoDeVeaux** [comment]
@tymor22 New Travis Error: in python 3.4 (2.7 is passing):

  File "pandas/lib.pyx", line 832, in pandas.lib.scalar_compare (pandas/lib.c:14847)
TypeError: unorderable types: str() > int()

File: inferelator_ng/design_response_translation.py [hunk 2]
```diff
@@ -23,8 +23,15 @@ def run(self, expression_mat, metadata_dataframe):
...
         for ch in special_char_dictionary.keys():
-            meta_data['condName']=meta_data['condName'].str.replace(ch,special_char_dictionary[ch])
-            meta_data['prevCol']=meta_data['prevCol'].str.replace(ch,special_char_dictionary[ch])
+            #meta_data['condName']=meta_data['condName'].str.replace(ch,special_char_dictionary[ch])
+            #meta_data['prevCol']=meta_data['prevCol'].str.replace(ch,special_char_dictionary[ch])
+            #meta_data['condName'][~meta_data['condName'].isnull()]=meta_data['condName'][~meta_data['condName'].isnull()].str.replace(ch,special_char_dictionary[ch])
+            #meta_data['prevCol'][~meta_data['prevCol'].isnull()]=meta_data['prevCol'][~meta_data['prevCol'].isnull()].str.replace(ch,special_char_dictionary[ch])
+            #need this edge case for passing micro test
+            if len(meta_data['condName'][~meta_data['condName'].isnull()]) > 0:
+                meta_data['condName']= meta_data['condName'].str.replace(ch,special_char_dictionary[ch])
```
Comment (NicoDeVeaux) [inline inferelator_ng/design_response_translation.py:32]: We're willing to sacrifice some readability in order to get the tests to pass. 

The next bit of work on design_response_translation.py will be focused on improving readability

**Comment by NicoDeVeaux** [comment]
Overall, it seems like this new code is just as hard to interpret as the original R code, but that's by design: it's a direct translation of the R code.  We're going to disentangle the translation part of this refactor from the re-thinking part. A future pull request will have the re-design of the design_and_response code.

**Comment by NicoDeVeaux** [comment]
Travis CI is passing, so 👍  from me to merge. 

We'll punt on getting CodeCov up higher until we move the BBSR R code to python.

**Review by NicoDeVeaux — APPROVED** [review]
